### PR TITLE
MGLP Stake refactoring 

### PR DESCRIPTION
**Pull request by Andrey-Onipchenko**

### CODE REVIEW

File: src/helpers/collateralsApy/getMegicGlpApy/constants.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import rewardReaderAbi from "@/helpers/collateralsApy/getMegicGlpApy/abi/RewardReader";
```
Comment (0xCalibur) [inline src/helpers/collateralsApy/getMegicGlpApy/constants.ts:1]: typo in the folder name

Comment (0xCalibur) [reply src/helpers/collateralsApy/getMegicGlpApy/constants.ts:1]: getMegicGlpApy

File: src/helpers/collateralsApy/index.js [hunk 2]
```diff
@@ -55,8 +55,11 @@ export const fetchTokenApy = async (pool) => {
   }
 
   if (chainId === 42161) {
-    if (pool.config.id === 2 || pool.config.id === 3)
-      return await getGlpApy(pool.config.id === 3);
+    if (pool.config.id === 2 || pool.config.id === 3) {
+      const response = await getMegicGlpApy(chainId);
+      if (pool.config.id === 2) return response.glpApy;
```
Comment (0xCalibur) [inline src/helpers/collateralsApy/index.js:60]: use be something like pool.config.collateralApyType instead and switch/case it

**Review by 0xCalibur — CHANGES_REQUESTED** [review]

File: src/helpers/collateralsApy/getMegicGlpApy/constants.ts (hunk 1 above)
Comment (Andrey-Onipchenko) [reply src/helpers/collateralsApy/getMegicGlpApy/constants.ts]: fixed typo

File: src/helpers/collateralsApy/index.js (hunk 2 above)
Comment (Andrey-Onipchenko) [reply src/helpers/collateralsApy/index.js:60]: This file is not related to the Stake refactoring. I'll fix it in the next updates

**Review by 0xCalibur — APPROVED** [review]

**Review by 0xhardrefresh — APPROVED** [review]

**Pull request merged**